.github/workflows/rust.yml

```diff
@@ -0,0 +1,36 @@
+---
+name: "Rust CI"
+on:
+  workflow_call:
+
+jobs:
+  build:
+    name: "Build"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Prepare dependencies
+        run: |
+          rustup install stable
+
+      - name: Extract crate information
+        id: crate_metadata
+        run: |
+          cargo metadata --no-deps --format-version 1 | jq -r '"name=" + .packages[0].name' | tee -a $GITHUB_OUTPUT
+          cargo metadata --no-deps --format-version 1 | jq -r '"version=" + .packages[0].version' | tee -a $GITHUB_OUTPUT
+
+      - name: Clippy
+        run: cargo clippy
+
+      - name: Build
+        run: cargo build --release
```

Comment:
This should be using https://github.com/rust-secure-code/cargo-auditable for the build, so that we can recover an SBOM from the binary using syft later